src/main.rs

```diff
@@ -177,7 +177,10 @@ impl BFInterp {
                 Constant::GetElementPtr(GetElementPtr { address, .. }) => match address.as_ref() {
                     Constant::GlobalReference { name, .. } => {
                         match self.gl_vars.get(name).unwrap() {
-                            Constant::Array { elements, .. } => self.arr_to_string(elements),
+                            Constant::Array { elements, .. } => {
+                                assert!(&name.to_string()[1..5] == ".str");
```

Comment:
Is the the only way to check if an array is a string. I saw something related to `ParameterAttributes` but am not sure if that's useful or not.

Reply:
ParameterAttributes for what?

Reply:
Nevermind, I thought you could get that info out of the parameter attributes that are passed to the call alongside the operands. But I don't think that contains the info I thought it did. It seems what you've done is AFAICS the only way.